src/features/SelectPSSARules.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import vscode = require("vscode");
+import { IFeature } from "../feature";
+import { LanguageClient, RequestType } from "vscode-languageclient";
+import { CheckboxQuickPickItem, CheckboxQuickPick } from "../checkboxQuickPick";
+
+export namespace GetPSSARulesRequest {
+    export const type: RequestType<any, any, void> = { get method(): string { return "powerShell/getPSSARules"; } };
+}
+
+export namespace SetPSSARulesRequest {
+    export const type: RequestType<any, any, void> = { get method(): string { return "powerShell/setPSSARules"; } };
+}
+
+class RuleInfo {
+    name: string;
+    isEnabled: boolean;
+}
+
+class SetPSSARulesRequestParams {
+    filepath: string;
+    ruleInfos: RuleInfo[];
+}
+
+export class SelectPSSARulesFeature implements IFeature {
+
+    private command: vscode.Disposable;
+    private languageClient: LanguageClient;
+
+    constructor() {
+        this.command = vscode.commands.registerCommand("PowerShell.SelectPSSARules", () => {
+            if (this.languageClient === undefined) {
+                return;
+            }
+
+            this.languageClient.sendRequest(GetPSSARulesRequest.type, null).then((returnedRules) => {
+                if (returnedRules == null) {
+                    vscode.window.showWarningMessage(
+                        "PowerShell extension uses PSScriptAnalyzer settings file - Cannot update rules.");
+                    return;
+                }
+                let options: CheckboxQuickPickItem[] = returnedRules.map(function (rule: RuleInfo): CheckboxQuickPickItem {
+                    return { name: rule.name, isSelected: rule.isEnabled };
+                });
+                CheckboxQuickPick.show(options, (updatedOptions) => {
+                    let filepath: string = vscode.window.activeTextEditor.document.uri.fsPath;
+                    let ruleInfos: RuleInfo[] = updatedOptions.map(
+                        function (option: CheckboxQuickPickItem): RuleInfo {
+                            return { name: option.name, isEnabled: option.isSelected };
+                        });
+                    let requestParams: SetPSSARulesRequestParams = {filepath, ruleInfos};
+                    this.languageClient.sendRequest(SetPSSARulesRequest.type, requestParams);
+                });
+            });
+        });
+    }
+
+    public setLanguageClient(languageclient: LanguageClient): void {
+        this.languageClient = languageclient;
```

Comment:
One more thing here:  I think your request for getting the PSSA rules needs to be done in setLanguageClient.  Since this method is called when the session is started, restarted, or switched to a new runtime, we'll need to re-query this list to make sure the list is up to date.

It makes sense to maintain the list of selected rules across sessions, though.  This might require getting the list of rules and comparing it against the current list of rules that have been set so that you can set those rules on the new session.  If this is too complicated to get done right now, we at least need to get the set of rules from the server when the LanguageClient changes.